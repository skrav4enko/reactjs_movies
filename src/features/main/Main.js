import React from 'react';
import './Main.scss';
import Header from '../../core/header/Header';
import Footer from '../../core/footer/Footer';
import SearchMovie from '../search-movie/SearchMovie';
import MovieDetail from '../movie-detail/MovieDetail';

import moviesData from '../../mock/data.json';

const movies = [...moviesData.data];

function Main(props) {
  return (
    <div className="main">
      <Header />
      <main className="content">
        <SearchMovie movies={movies} />
        {/* <MovieDetail movie={movies[0]} movies={movies} /> */}
      </main>
      <Footer />
    </div>
  );
}

export default Main;
