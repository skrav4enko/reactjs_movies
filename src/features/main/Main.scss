.main {
  position: relative;
  background-color: #232323;
  color: #ffffff;
  height: 100%;
  display: grid;
  grid-template-rows: 1fr 70px;

  .content {
    display: flex;
    flex-direction: column;
  }
}
