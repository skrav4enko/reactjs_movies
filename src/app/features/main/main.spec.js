/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import Main from '.';

describe('Main', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Main />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handke sort value', () => {
    const testSortBy = 'vote_average';
    const wrapper = shallow(<Main />);
    const instance = wrapper.instance();
    instance.setState({
      movies: [{
        id: 338970,
        title: 'Tomb Raider',
        tagline: 'Her legend begins',
        vote_average: 6.2,
        vote_count: 817,
        release_date: '2018-03-08',
        poster_path: 'https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg',
        overview:
          'Lara Croft, the fiercely independent daughter of a missing adventurer.',
        budget: 94000000,
        revenue: 126025000,
        genres: ['Action', 'Adventure'],
        runtime: 118,
      }]
    });
    instance.handleSort(testSortBy)

    expect(wrapper.state().sortBy).toBe(testSortBy);
  });

  it('should filter by value', () => {
    const testFilter = 'Croft';
    const wrapper = shallow(<Main />);
    const instance = wrapper.instance();
    instance.setState({
      movies: [{
        id: 338970,
        title: 'Tomb Raider',
        tagline: 'Her legend begins',
        vote_average: 6.2,
        vote_count: 817,
        release_date: '2018-03-08',
        poster_path: 'https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg',
        overview:
          'Lara Croft, the fiercely independent daughter of a missing adventurer.',
        budget: 94000000,
        revenue: 126025000,
        genres: ['Action', 'Adventure'],
        runtime: 118,
      }, {
        id: 338971,
        title: 'Lara Croft',
        tagline: 'Her legend begins',
        vote_average: 6.2,
        vote_count: 817,
        release_date: '2018-03-08',
        poster_path: 'https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg',
        overview:
          'Lara Croft, the fiercely independent daughter of a missing adventurer.',
        budget: 94000000,
        revenue: 126025000,
        genres: ['Action', 'Adventure'],
        runtime: 118,
      }]
    });
    instance.handleFilter(testFilter)

    expect(wrapper.state().filteredMovies).toHaveLength(1);
  });

  it('should filter empty by value', () => {
    const testFilter = '';
    const wrapper = shallow(<Main />);
    const instance = wrapper.instance();
    instance.setState({
      movies: [{
        id: 338970,
        title: 'Tomb Raider',
      }, {
        id: 338971,
        title: 'Lara Croft'
      }]
    });
    instance.handleFilter(testFilter)

    expect(wrapper.state().filteredMovies).toHaveLength(2);
  });


  it('should navigate by value', () => {
    const testId = 338970;
    const wrapper = shallow(<Main />);
    const instance = wrapper.instance();
    instance.setState({
      movies: [{
        id: 338970,
        title: 'Tomb Raider',
      }, {
        id: 338971,
        title: 'Lara Croft'
      }]
    });
    instance.handleNavigate(testId)

    expect(wrapper.state().selectedMovie.title).toBe('Tomb Raider');
  });
});

