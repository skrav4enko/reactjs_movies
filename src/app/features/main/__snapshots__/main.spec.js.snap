// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `
<div
  className="main"
>
  <Header />
  <main
    className="content"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/film/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="**"
      />
    </Switch>
  </main>
  <Footer />
</div>
`;
